File: 1047-remove-all-adjacent-duplicates-in-string/1047-remove-all-adjacent-duplicates-in-string.js
/**
 * @param {string} s
 * @return {string}
 */
var removeDuplicates = function(s) {
    const res = []
    for(let i=0;i<s.length;i++){
        if(res.length && res[res.length -1] === s[i]){
            res.pop()
        }
        else{
            res.push(s[i])
        }
    }
    return res.join("")
};

File: 1047-remove-all-adjacent-duplicates-in-string/1047-remove-all-adjacent-duplicates-in-string.py
class Solution(object):
    def removeDuplicates(self, s):
        """
        :type s: str
        :rtype: str
        """
        stack=[]
        for i in s:
            if stack and stack[-1] == i:
                stack.pop()
            else:
                stack.append(i) 

        res = "".join(stack)        
        return res   
            
        